### ISSUE TITLE
Ability to share by tag

### ISSUE THREAD

**Issue by Bugsbane** [owncloud/core#13168]
As mentioned in https://github.com/owncloud/core/issues/2368 , it would be useful to be able to share resources by tag. For example, I'd like to be able to go through and tag photos and calendar events with "Family" and have them be shared with my family members (users in the "Family" group). Similarly, I'd like to be able to tag bookmarks, contacts and documents with a project tag, and have them all be shared with the team members involved with that project.

I'm not sure how the UI would work, but I imagine something like a "Tags" app, which would let me:
1. See all currently used tags
2. Share tags in the same way as how we do with files currently
3. List all assets with the given tag
4. Being able to rename / delete the tags.

**Comment by mbiebl** [comment]
Completely agree. Atm, the only way to do something like that, is by using subfolders and moving files around. If you want to share the same files to different groups, you might have to copy the file.

I usually have my photos in a single huge folder, so sharing by tagging would be a huge plus.

I would like to see this mechanism extended for the Pictures app as well. I'd like to tag pictures and create a gallery from it. Again, atm this is only really possible by using (sub)folders.

**Comment by oparoz** [comment]
This is crucial for any professional environment. You want one copy of each file, not several dispatched to multiple folders. It's a nightmare to keep track of revisions that way.

**Comment by moritzheiber** [comment]
:+1: for this!

**Comment by Bugsbane** [comment]
Now that v9 with tagging is out, I'm hoping this can be revisited... I'd really like to be able to just tag pictures with who's in them and automatically have photos of my kids shared with family overseas, and likewise, automatically have files tagged with project specific tags, shared with work colleagues, without having to go back and manually create separate shares for each and every file.

**Comment by joephein** [comment]
This would be a great feature. I would very much like to see this in ownCloud.

**Comment by Bugsbane** [comment]
To anyone supporting this ( @mbiebl @oparoz @moritzheiber @joephein etc.), thank you. As much as possible, if you're mainly commenting to voice your support, consider using the smiley face icon > "Add your reaction" in the top right corner of the original post so devs can see the number of people supporting it in one place without getting their notifications spammed.

**Comment by ronnystandtke** [comment]
Here comes another use case:
I record my lessons at university and put them into a corresponding lesson folder that also contains all the other lesson specific documents. Now I wanted to share all the recorded videos with a friend who has no account on my ownCloud server. My ideal solution would be to just tag the video files and create a folder shared by link that contains only files with this tag so that my friend could access and download them via a simple URL.
Unfortunately, it looks like with the current ownCloud version I have to copy all video files into a separate folder and share this folder by link. Because ownCloud doesn't support symlinks I am going to waste an awful amount of disk space by duplicating all the video files...
Long story short: I would love to have this feature added to ownCloud.

**Comment by PVince81** [comment]
@pmaier1

**Comment by PVince81** [comment]
Would this fit with the "project rooms" idea ? Here the tags would be used to mark stuff to be added to projects.
